ibis/backends/base/sqlglot/__init__.py

```diff
@@ -151,6 +151,17 @@ def _get_schema_using_query(self, query: str) -> sch.Schema:
         """Return an ibis Schema from a backend-specific SQL string."""
         return sch.Schema.from_tuples(self._metadata(query))
 
+    def _get_sql_string_view_schema(self, child: ir.Table, query: str) -> sch.Schema:
+        # abort if somehow this was called with a non-view
+        assert isinstance(child.op(), ops.View), type(child.op())
+
+        obj = self._to_sqlglot(child)
+        dialect = self.compiler.dialect
+        parsed = sg.parse_one(query, read=dialect)
```

Comment:
Could we avoid the parsing here? I like to think `.sql()` as the escape hatch if something is not supported by ibis or sqlglot :)

Reply:
Not really, there could be multiple `with`s that need to be added to the parsed query.